contracts/StakingPoolV2.sol

```diff
@@ -130,8 +132,37 @@ contract StakingPoolV2 is IStakingPoolV2, StakedElyfiToken {
     _claim(msg.sender);
   }
 
-  // TODO Implement `migrate` function to send an asset to the next staking contract
-  
+  /// @notice Migrate the amount of principal to the current round and transfer the rest principal to the caller
+  function migrate() external override {
+    if (_poolData.isOpened == true) revert Opened();
+    if (_nextContractAddr == address(0)) revert NotSetContractAddr();
+    if (_poolData.userPrincipal[msg.sender] == 0) revert ZeroPrincipal();
+    uint256 amount = _poolData.userPrincipal[msg.sender];
+
+    
+    // Claim reward
+    if (_poolData.getUserReward(msg.sender) != 0) {
+      _claim(msg.sender);
+    }
+
+    _poolData.updateStakingPool(msg.sender);
+
+    // Migrate user, total principal 
+    _poolData.userPrincipal[msg.sender] -= amount;
+    _poolData.totalPrincipal -= amount;
+
+    // Migrate
+    _migrateTo(_nextContractAddr, amount);
+
+    // Call next contract
+    INextStakingPool nextContract = INextStakingPool(_nextContractAddr);
+
+    // Migrate next contract of user, total principal
+    nextContract.setPreviousPoolData(msg.sender, amount);
+
+    emit Migrate(msg.sender);
```

Comment:
For making this event to be more practical, I'd recommend adding other parameters to the `Migrate` event